content_cafe.py

```diff
@@ -73,22 +74,35 @@ class ContentCafeAPI(object):
     excerpt_url = BASE_URL + "ContentCafeClient/Excerpt.aspx?UserID=%(userid)s&Password=%(password)s&ItemKey=%(isbn)s"
     author_notes_url = BASE_URL + "ContentCafeClient/AuthorNotes.aspx?UserID=%(userid)s&Password=%(password)s&ItemKey=%(isbn)s"
 
-    def __init__(self, db, mirror, user_id=None, password=None, uploader=None,
+    log = logging.getLogger("Content Cafe API")
+
+    @classmethod
+    def from_config(cls, _db, mirror, **kwargs):
+        integration = ExternalIntegration.lookup(
+            _db, ExternalIntegration.CONTENT_CAFE,
+            ExternalIntegration.METADATA_GOAL
+        )
+
+        if not integration or not (integration.username and integration.password):
+            raise ValueError('Content Cafe not properly configured')
```

Comment:
Can you use CannotLoadConfiguration here? While starting up the circ manager from scratch I decided it's be easier if we changed CannotLoadConfiguration to start referring to configuration items stored in the database, not just in the config file.